lib/Controller/SignalingController.php

```diff
@@ -461,19 +461,22 @@ protected function getInputStream(): string {
 	 * https://nextcloud-spreed-signaling.readthedocs.io/en/latest/standalone-signaling-api-v1/#backend-requests
 	 *
 	 * @PublicPage
+	 * @BruteForceProtection(action=signalingSecret)
 	 *
 	 * @return DataResponse
 	 */
 	public function backend(): DataResponse {
 		$json = $this->getInputStream();
 		if (!$this->validateBackendRequest($json)) {
-			return new DataResponse([
+			$response = new DataResponse([
 				'type' => 'error',
 				'error' => [
 					'code' => 'invalid_request',
 					'message' => 'The request could not be authenticated.',
 				],
 			]);
+			$response->throttle();
```

Comment:
The delay must be reset if the backend request is valid. Otherwise if a wrong secret is set in the Talk administration settings and a conversation is opened the backend requests will be throttled, but as the delay is not reset then even after fixing the secret it will not be possible to join a conversation, because the request will get delayed longer than the reconnection timeout and the participant will just try to connect again and again.

Reply:
it's only one time where the delay is longer, after that a 429 is being returned.
But yeah makes sense to reset

Reply:
Would it be possible to automatically reset the bruteforce attempts when a new secret is set? The IP address to reset could be resolved from the domain of the configured HPB server. I do not know if that could fail in some cases, though (for example if the HPB is in the same local network as the Nextcloud server and the requests were throttled using the local IP, but the domain of the configured HPB server resolves to the public IP).

Alternatively (or in addition to), maybe a message can be shown in the UI to reset the bruteforce attempts if the HPB requests are throttled. It could be done as part of https://github.com/nextcloud/spreed/issues/7164 too instead of now.

Reply:
Automation is though at least as long as there is no way to reset only by action. I don't feel like making additional effort to reverse engineer the IP of the HPB.
But we can add a message in the UI in a follow up

Reply:
> I don't feel like making additional effort to reverse engineer the IP of the HPB.

It would not need to be bullet-proof, just a help for common cases. Just copy-pasting and adjusting [`DnsPinMiddleware::dnsResolve`](https://github.com/nextcloud/server/blob/84d9b17dc7105d32b84e717966730fc00c4ed5be/lib/private/Http/Client/DnsPinMiddleware.php#L66) should be enough. But OK, if there is at least a message explaining what could be the problem if connections fail fine by me.